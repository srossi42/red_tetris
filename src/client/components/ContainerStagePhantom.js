import React, { Fragment } from 'react'

import styled from 'styled-components'

import StagePhantom from './StagePhantom';

const container = {
    display: "flex",
    alignItems: 'stretch',
    flexWrap: 'wrap',
    width: '50%',
    height: '80%',
    backgroundColor: 'grey',
}

const TextInput = styled.h2`
    font-family: Montserrat;
    width: 100%;
    text-align: center;
    color: ${props => props.color};
`

const ContainerStagePhantom = ({boardAdversary, name}) => {
    return (
        <div style={container}>
            {boardAdversary ? boardAdversary.map((adversary, index) => {
                if (adversary.name !== name) {
                    return (
                    <Fragment key={index}>
                        <StagePhantom stage={adversary.board}/>
                        <TextInput color='#FF3333'>{adversary.name}</TextInput>
                    </Fragment>
                    )
                } else return
            }) : null}
        </div>
    )
}

export default ContainerStagePhantom