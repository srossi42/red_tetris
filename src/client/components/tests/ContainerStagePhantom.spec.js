import React from 'react'
import { shallow } from 'enzyme';
import ContainerStagePhantom from '../ContainerStagePhantom';
import { createStage } from '../../gameHelper'
import toJson from "enzyme-to-json"


describe('<ContainerStagePhantom/> Component', () => {
  it('rendering correctly ContainerStagePhantom', () => {
    const boardArray = [
      {
        board: createStage, 
        name: "lol", 
        end: true
      },
      {
        board: createStage, 
        name: 'hihi', 
        end: false
      }
    ]
    let wrapper = shallow(
      <ContainerStagePhantom boardAdversary={boardArray} name="hihi"/>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
