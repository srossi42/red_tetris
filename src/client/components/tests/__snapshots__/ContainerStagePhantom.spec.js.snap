// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerStagePhantom/> Component rendering correctly ContainerStagePhantom 1`] = `
<Fragment>
  <div
    key="0"
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "grey",
        "display": "flex",
        "flex": "1",
        "flexWrap": "wrap",
        "height": "80%",
        "margin": "10px",
        "maxWidth": "13.5vh",
        "position": "relative",
      }
    }
  >
    <StagePhantom
      stage={[Function]}
    />
    <GameOver
      win={false}
    />
    <styled.h2
      color="#FF3333"
    >
      lol
    </styled.h2>
  </div>
</Fragment>
`;
